reframe/core/parameters.py

```diff
@@ -124,9 +124,17 @@ def extend(self, cls):
 
         local_param_space = getattr(cls, self.local_namespace_name)
         for name, p in local_param_space.items():
-            self.params[name] = (
-                p.filter_params(self.params.get(name, ())) + p.values
-            )
+            try:
+                self.params[name] = (
+                    tuple(p.filter_params(self.params.get(name, ()))) +
+                    p.values
+                )
+            except TypeError:
```

Comment:
Here we are masking potential `TypeError` raised in the user callable. I'd rather use the `is_trivially_callable()` utility from #2083 and raise a separate `TypeError` (I don't think that `ReframeSyntaxError` is needed here, since the error is triggered immediately by the offending user line).

Reply:
Good point. I've split this into multiple steps to be able to raise the appropriate error on each step. Regarding the use of the `is_trivially_callable()` utility, that can be added instead in the `TestParam`'s `__init__()` once #2083 is in.